fetch.bs

```diff
@@ -471,6 +685,9 @@ each other by 0x2C 0x20, in order.
      "<code>text/plain</code>", then return false.
     </ol>
 
+    <p class=warning>This intentionally does not use <a>extract a MIME type</a> as that algorithm is
+    rather forgiving and servers are not expected to implement it.
```

Comment:
I find this warning confusing. Why are we talking about what servers should implement here? Isn't this part of the processing model for the browser?

Reply:
Yeah, but the security problem is a mismatch between the browser and a server and therefore this parser is more conservative.

Reply:
I still don't really understand. Why do we have a second parser on the client, to solve a client/server mismatch?

Maybe a better way to state this (either to help me understand here, or in the spec) would be with an example of what would go wrong if we used extract a MIME type.

Reply:
If we use the "liberal" parser for this the client might end up with text/plain but the server with application/json or some such. This would end up rejecting such input.

Reply:
So the mismatch scenario is: client uses the Headers class to add content-type: someHardToParseThingWhichTheyIntendToBeTextPlain, and does a fetch() post to the server, which uses a different parser and ends up with application/json?

If so, I'll note that's not really an attack (the client could just send application/json directly). So maybe this is more of a note than a warning. In either case, adding such an example would be helpful.

Reply:
How could the client send application/json without a preflight?

Reply:
I see, text/plain is explicitly allowed! This example would indeed be great to add.

Reply:
Added.